clientv3/integration/kv_test.go

```diff
@@ -179,3 +179,104 @@ func TestKVRange(t *testing.T) {
 		}
 	}
 }
+
+func TestKVDeleteRange(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	kv := clientv3.NewKV(clus.RandClient())
+
+	keySet := []string{"a", "b", "c", "c", "c", "d", "e", "f"}
+	for i, key := range keySet {
+		if _, err := kv.Put(key, "", lease.NoLease); err != nil {
+			t.Fatalf("#%d: couldn't put %q (%v)", i, key, err)
+		}
+	}
+
+	tests := []struct {
+		key, end string
+		delRev   int64
+	}{
+		{"a", "b", int64(len(keySet) + 2)}, // delete [a, b)
+		{"d", "f", int64(len(keySet) + 3)}, // delete [d, f)
+	}
+
+	for i, tt := range tests {
+		dresp, err := kv.DeleteRange(tt.key, tt.end)
+		if err != nil {
+			t.Fatalf("#%d: couldn't delete range (%v)", i, err)
+		}
+		if dresp.Header.Revision != tt.delRev {
+			t.Fatalf("#%d: dresp.Header.Revision got %d, want %d", i, dresp.Header.Revision, tt.delRev)
+		}
+		grespKey, err := kv.Get(tt.key, 0)
```

Comment:
we do not have to test get here. we should assume the server side do the current thing. but it does not hurt either.